compiler/rustc_typeck/src/check/fn_ctxt/checks.rs

```diff
@@ -1137,6 +1137,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             } else if error.obligation.cause.span == call_sp {
                 // Make function calls point at the callee, not the whole thing.
                 if let hir::ExprKind::Call(callee, _) = expr.kind {
+                    let parent_code = error.obligation.cause.clone_code();
+                    *error.obligation.cause.make_mut_code() =
+                        ObligationCauseCode::FunctionCallObligation {
+                            call_hir_id: expr.hir_id,
+                            parent_code,
+                        };
```

Comment:
I'm not sure what this PR really does.
Which obligation code was previously used for this case?
Can this new code make any observable changes in tests?
Also `call_hir_id` is set here but never used.

Reply:
>That's the problem: almost any obligation code could have been previously used for this case.

Why is that a problem though?
Why is it necessary to wrap the parent obligation into something instead of using it directly? Especially if it is not observable from the outside.

Reply:
I’m not sure.

@estebank what else did you want done here?